### ISSUE TITLE
IllegalStateException for ch.qos.logback.classic.util.LogbackMDCAdapter

### ISSUE THREAD

**Issue by CodingFabian** [kamon-io/Kamon#790]
Hello,
a customer of ours reported this exception

```
ERROR 2020-05-04 14:41:57 Logger : Error => ch.qos.logback.classic.util.LogbackMDCAdapter with message size = 2. Class loader: sun.misc.Launcher$AppClassLoader@70dea4e: java.lang.IllegalStateException: size = 2 
at kanela.agent.libs.net.bytebuddy.matcher.FilterableList$AbstractBase.getOnly(FilterableList.java:139) 
at kanela.agent.libs.net.bytebuddy.implementation.MethodDelegation$ImplementationDelegate$ForField$WithInstance.resolve(MethodDelegation.java:1032) 
at kanela.agent.libs.net.bytebuddy.implementation.MethodDelegation$ImplementationDelegate$ForField.compile(MethodDelegation.java:952) 
at kanela.agent.libs.net.bytebuddy.implementation.MethodDelegation.appender(MethodDelegation.java:601) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.MethodRegistry$Handler$ForImplementation.compile(MethodRegistry.java:233) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.MethodRegistry$Handler$ForImplementation.compile(MethodRegistry.java:205) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.MethodRegistry$Default$Prepared.compile(MethodRegistry.java:740) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining$WithFullProcessing$RedefinitionClassVisitor.visit(TypeWriter.java:4302) 
at kanela.agent.libs.net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:533) 
at kanela.agent.libs.net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:400) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3397) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:1933) 
at kanela.agent.libs.net.bytebuddy.dynamic.scaffold.inline.RebaseDynamicTypeBuilder.make(RebaseDynamicTypeBuilder.java:245) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$Transformation$Simple$Resolution.apply(AgentBuilder.java:10123) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer.doTransform(AgentBuilder.java:10494) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer.transform(AgentBuilder.java:10457) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer.access$1500(AgentBuilder.java:10223) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer$LegacyVmDispatcher.run(AgentBuilder.java:10833) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer$LegacyVmDispatcher.run(AgentBuilder.java:10780) 
at java.security.AccessController.doPrivileged(Native Method) 
at kanela.agent.libs.net.bytebuddy.agent.builder.AgentBuilder$Default$ExecutingTransformer.transform(AgentBuilder.java:10380) 
```

I could not figure out where in your instrumentation code this is caused.

**Comment by CodingFabian** [comment]
It is worth noting that we do instrument this method as well using bytebuddy.

We change `getPropertyMap` like this
```
  @Advice.OnMethodExit(suppress = Throwable.class)
  private static void exit(@Tag int tag, @Advice.Return(readOnly = false) Map stringStringMap) {
    Object traceId = InstanaTraceDispatcherV1.find(tag).call(UNUSED);
    if (traceId != null) {
      HashMap hashMap = new HashMap();
      if (stringStringMap != null) {
        hashMap.putAll(stringStringMap);
      }
      hashMap.put("instana.trace.id", traceId);
      stringStringMap = hashMap;
    }
  }
```

**Comment by raphw** [comment]
Seems like `MethodDelegation$ImplementationDelegate$ForField$WithInstance.resolve(MethodDelegation.java:1032) ` fails to account for an edge case where a field with the same name but with different types (those are distinct for the JVM) exists in the class. Question is: who put this extra field in there? Advice does not add fields, normally. Is the delegation added twice, maybe?

**Comment by raphw** [comment]
I think this issue is already solved in Byte Buddy 1.10. If you can update, I think this compatibility might just disappear.

**Comment by dpsoft** [comment]
Hi guys, thanks for pointing this, We will update our agent to the latest Byte Buddy version.

**Comment by dpsoft** [comment]
closed with https://github.com/kamon-io/kanela/commit/e83b05d50eb3e98b7a043c98317026525ab58c3b

**Comment by dpsoft** [comment]
closed with kamon-io/kanela@e83b05d
